tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/GitHubService.cs

```diff
@@ -311,5 +324,49 @@ public async Task<Issue> GetIssueAsync(string repoOwner, string repoName, int is
                 throw;
             }
         }
+
+        /// <summary>
+        /// Appends release plan and spec PR links to the body of a pull request and updates the PR on GitHub.
+        /// </summary>
+        /// <param name="repoOwner">The owner of the repository.</param>
+        /// <param name="repoName">The name of the repository.</param>
+        /// <param name="prNumber">The pull request number.</param>
+        /// <param name="releasePlanLink">The link to the release plan.</param>
+        /// <param name="specPrLink">The link to the spec pull request.</param>
+        public async Task AddReleasePlanInfoInSdkAsync(string repoOwner, string repoName, int prNumber, string releasePlanLink, string specPrLink, string workItemLink, string apiVersion)
+        {
+            var pr = await GetPullRequestAsync(repoOwner, repoName, prNumber);
+            if (pr == null)
+            {
+                logger.LogError($"Failed to fetch pull request {repoOwner}/{repoName}#{prNumber}");
+                return;
+            }
+
+            var links = $"Release Plan: {releasePlanLink}";
+            if (!string.IsNullOrEmpty(workItemLink))
+            {
+                links += $"\nWork Item Link: {workItemLink}";
+            }
+            if (!string.IsNullOrEmpty(specPrLink))
+            {
+                links += $"\nSpec Pull Request: {specPrLink}";
+            }
+            if (!string.IsNullOrEmpty(apiVersion))
+            {
+                links += $"\nSpec API version: {apiVersion}";
+            }
+
+            // Check if the PR body already contains the release plan/spec PR info
+            if (!string.IsNullOrEmpty(pr.Body) && pr.Body.Contains(links))
```

Comment:
alternate approach check this is to add header in the PR body like `## Release plan details` and just check if PR body contains this header instead of checking entire content.